src/ndarray.jl

```diff
@@ -656,13 +656,21 @@ Matrix (2D NDArray) multiplication.
 
 Elementwise divide a scalar or an `NDArray` of the same shape from `dst`. Inplace updating.
 """
-function div_from!(dst::NDArray{T}, arg::NDArrayOrReal) where {T}
+function div_from!(dst::NDArray, arg::NDArrayOrReal)
   @assert dst.writable
   if isa(arg, Real)
-    _div_scalar(dst, scalar = convert(T, arg), out = dst)
+    _div_scalar(dst, scalar = arg, out = dst)
   else
     _div(dst, arg, out = dst)
   end
+  dst
+end
+
+function div_from!(dst::NDArray{T}, arg::Real) where {T<:Integer}
+  @assert dst.writable
+  @assert(round(T, arg) != zero(T), "Integer divided by zero")
```

Comment:
It seems Julia built-in array throws `InexactError` instead of assertion error when assigning out of bound floating point to integer arrays. Shall we follow the convention?

Reply:
In Julia's Array, dividing-by-zero will return `Inf`,
 and the type of Array become `Array{Float64}`
```julia
julia> [1, 2, 3] / 0                                                                                                                    
3-element Array{Float64,1}:                                                                                                             
 Inf                                                                                                                                    
 Inf                                                                                                                                    
 Inf
```

But, if I try to do this in libmxnet. libmxnet just raise segfault, and crash julia process.
That's why I try to do this kind of protection.
```julia
julia> using MXNet                                                                                                                      
INFO: Recompiling stale cache file /home/iblis/.julia/lib/v0.6/MXNet.ji for module MXNet.                                               
                                                                                                                                        
julia> mx.NDArray([1, 2, 3]) / 0                                                                                                        
zsh: segmentation fault (core dumped)  julia
```